applications/sudokumb/Counters/Counters.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2018 Google LLC.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Sudokumb
+{
+    public interface ICounter
+    {
+        void Increase(long amount);
+        long Count { get; }
+    }
+
+    public class UnsynchronizedCounter : ICounter
+    {
+        long _count = 0;
+        public long Count => _count;
+        public void Increase(long amount)
+        {
+            _count += amount;
+        }
+    }
+
+    public class LockingCounter : ICounter
+    {
+        long _count = 0;
+        readonly object _thisLock = new object();
+
+        public long Count
+        {
+            get
+            {
+                lock (_thisLock)
```

Comment:
In the benchmark, the get Count operation happens only 3 times per second, while Increase() is called 10,000 to 100,000 times per second, so I don't think it will affect performance of the benchmark.  I'd rather stick with the simpler code.